releases_test.go

```diff
@@ -0,0 +1,206 @@
+					timestamp:  "2018-02-02T00:00:00.000000Z",
+					version:    "1.1.1",
+					deprecated: true,
+				},
+			},
+			expectedOutput: []Release{
+				{
+					timestamp: "2018-02-08T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+			},
+		},
+		{
+			name: "case 3: test with three same versions in multiple Releases",
+			input: []Release{
+				{
+					timestamp:  "2018-02-08T00:00:00.000000Z",
+					version:    "1.1.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-03-22T00:00:00.000000Z",
+					version:   "1.2.0",
+					wip:       true,
+				},
+				{
+					timestamp:  "2018-02-02T00:00:00.000000Z",
+					version:    "1.0.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-02-18T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+				{
+					timestamp:  "2018-02-12T00:00:00.000000Z",
+					version:    "1.1.1",
+					deprecated: true,
+				},
+			},
+			expectedOutput: []Release{
+				{
+					timestamp:  "2018-02-02T00:00:00.000000Z",
+					version:    "1.0.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-02-18T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+				{
+					timestamp: "2018-03-22T00:00:00.000000Z",
+					version:   "1.2.0",
+					wip:       true,
+				},
+			},
+		},
+		{
+			name: "case 4: test with two Releases with same version where older is active",
+			input: []Release{
+				{
+					timestamp: "2018-02-08T00:00:00.000000Z",
+					version:   "1.1.1",
+					wip:       true,
+				},
+				{
+					timestamp: "2018-02-02T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+			},
+			expectedOutput: []Release{
+				{
+					timestamp: "2018-02-02T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+			},
+		},
+		{
+			name: "case 5: test with three same versions in multiple Releases where middle one of duplicates is active",
+			input: []Release{
+				{
+					timestamp:  "2018-02-08T00:00:00.000000Z",
+					version:    "1.1.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-03-22T00:00:00.000000Z",
+					version:   "1.2.0",
+					wip:       true,
+				},
+				{
+					timestamp:  "2018-02-02T00:00:00.000000Z",
+					version:    "1.0.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-02-18T00:00:00.000000Z",
+					version:   "1.1.1",
+					wip:       true,
+				},
+				{
+					timestamp: "2018-02-12T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+			},
+			expectedOutput: []Release{
+				{
+					timestamp:  "2018-02-02T00:00:00.000000Z",
+					version:    "1.0.1",
+					deprecated: true,
+				},
+				{
+					timestamp: "2018-02-12T00:00:00.000000Z",
+					version:   "1.1.1",
+				},
+				{
+					timestamp: "2018-03-22T00:00:00.000000Z",
+					version:   "1.2.0",
+					wip:       true,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			output := deduplicateReleases(tc.input)
+
+			if !reflect.DeepEqual(output, tc.expectedOutput) {
+				fmt.Printf("%s: This is what I got: \n", tc.name)
```

Comment:
This is not aligned with our `fmt` style regarding failure message, but debugging from `("%#v", output)` is just impossible.